packages/syft/src/syft/service/sync/sync_service.py

```diff
@@ -135,8 +146,22 @@ def add_permissions_for_item(
                 )
                 job_store.add_permission(permission)
 
+    def add_storage_permissions_for_item(
+        self,
+        context: AuthedServiceContext,
+        item: SyftObject,
+        permissions_other: Set[UID],
+    ) -> None:
+        _id = item.id.id
+        permissions = [
+            StoragePermission(uid=_id, node_uid=p) for p in permissions_other
```

Comment:
is this correct? we should not always give the same permissions on this side as the other node right?

Reply:
do this logic client side